regular_expressions_boundaries_anchors/regular_expressions_boundaries_anchors.md

```diff
@@ -0,0 +1,425 @@
+
+</div>
+
+### Except: Another Use of the Caret 
+
+On the previous page, we warned you about a potential pitfall:
+
+> To correctly use `^` as the start-of-string token, you must not enclose it in square brackets.  If `^` appears inside square brackets, it will be intepreted as doing something quite different, as you are about to discover.
+
+Let's look at three ways we can use the caret (`^`).
+
+Start-of-string Caret
+-----
+
+If we want to indicate the start of a string, we add the caret at the beginning of our regular expression, being sure to not enclose it in square brackets.  We see an example of this in the regular expression `^(?:WT|PBMC)\-\d{5,6}`, which includes a start of string character followed by a non-capturing group with two alternatives, a hyphen, and then five to six digits.
+
+Except-for Caret
+------
+
+We can also use the caret symbol in an "except for" clause in our regular expression.  For example, what if at a given position in a pattern, any character is possible except for a hash mark (#)?  We can say "except for" by using `^` as the first character within square brackets: `[^#]`.  
+
+We can add multiple characters that should not appear within the square brackets.  If any character except for  #, 5, ., and % are permitted at a given location in a string, we could express that with the regular expression list `[^#5\.%]`. For example, `[ABXYZ]` means "any of these: A, B, X, Y, or Z," while `[^ABXYZ]` means "anything that **isn't** one of these: A, B, X, Y, and Z."  
+
+<div class = "important">
+<b style="color: rgb(var(--color-highlight));">Important note</b><br>
+
+To use `^` as "except for", it has to immediately follow an opening square bracket, and be followed by the characters it is indicating aren't acceptable.  You can't, for example, say "all the digits except for 7" by typing `\d^7` or `[\d^7]`.  One way to do that would be `[012345689]`, and another would be `[0-689]`.  
+
+In some regular expression environments, you can do some advanced practices such as creating an **intersection** between two sets (such as the set of all digits and the set of all symbols except for 7), but the way to do this is different in different languages and regular expressions environments, so we won't go into detail about that here.
+
+Similarly, if you wanted to define a list with all uppercase letters except F and L, the most universal way to accomplish that is `[A-EG-KM-Z]`, but you'll sometimes see other, system-specific regular expressions that go beyond the scope of this module.
+
+</div>
+
+Literal Caret
+---
+
+If you're looking for the actual symbol `^` in your text, you'll have to **escape** it by adding a backslash right before the caret: `\^`.  As a reminder, that backslash is useful for all **metacharacters** in regular expressions.  You'd do the same thing for the end-of-string character: `\$` is how you look for a literal dollar sign / money symbol.
+
+### End-of-string Anchoring 
+
```

Comment:
Ok, here is a proposed update that will satisfy my grammar prescriptivism for published materials (I try to be a descriptivist with language but punctuation is more of struggle).

```
As a reminder, the symbol for end-of-string in a regular expression is `$`.  We can use this symbol to look for specific patterns at the end of strings.  For example, we could look for specific typos that tend to appear at the end of strings: periods that follow something other than a letter or number.  Here are some examples of acceptable grammar and typos:

**Acceptable**

* This is with respect to record 117.
* Participants were not clear on pre-registration next steps.


**Typos**

* Reported she "did not receive the sample".
* A chi-squared test would not be robust . 

The pattern we're looking for is "anything that is not a letter or digit, followed by a period," or `[^\w]\.$`.  If you want to see this in action, you can [try it on Regex 101](https://regex101.com/r/4cZn0e/1).
```

Reply:
I like it!